src/nvim/os/event.h

```diff
@@ -8,9 +8,15 @@
 #include "nvim/os/job_defs.h"
 #include "nvim/os/time.h"
 
-// Poll for events until a condition is true or a timeout has passed
+void ui_busy_start(void);
+void ui_busy_stop(void);
+
+// Poll for events until a condition or timeout
 #define event_poll_until(timeout, condition)                                 \
   do {                                                                       \
+    if (timeout < 0 || timeout > 100) {                                      \
```

Comment:
The idea is to only change the busy state if the poll can block for more than 100 ms